### PR TITLE
listener interface

### PR DESCRIPTION
**Pull request by critical27**
listener interface, based on https://github.com/vesoft-inc/nebula-common/pull/279, rely on https://github.com/vesoft-inc/nebula-storage/pull/180

### CODE REVIEW

**Comment by bright-starry-sky** [comment]
Could you resolved the conflicts? Thanks.

**Comment by critical27** [comment]
> Could you resolve the conflicts? Thanks.

done

File: src/executor/admin/ListenerExecutor.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    AddListenerExecutor(const PlanNode *node, QueryContext *qctx)
+        : Executor("AddListenerExecutor", node, qctx) {}
+
+    folly::Future<Status> execute() override;
+};
+
+class RemoveListenerExecutor final : public Executor {
+public:
+    RemoveListenerExecutor(const PlanNode *node, QueryContext *qctx)
+            : Executor("RemoveListenerExecutor", node, qctx) {}
```
Comment (bright-starry-sky) [inline src/executor/admin/ListenerExecutor.h:26]: alignment

File: src/parser/parser.yy [hunk 2]
```diff
@@ -2508,6 +2512,24 @@ drop_snapshot_sentence
     }
     ;
 
+add_listener_sentence
+    : KW_ADD KW_LISTENER name_label host_list {
```
Comment (bright-starry-sky) [inline src/parser/parser.yy:2516]: Would it be better to replace name_label to KW_ELASTICSEARCH ?

File: src/validator/AdminValidator.cpp [hunk 3]
```diff
@@ -221,6 +221,63 @@ Status ShowSnapshotsValidator::toPlan() {
...
+
+Status RemoveListenerValidator::validateImpl() {
+    auto sentence = static_cast<RemoveListenerSentence*>(sentence_);
+    auto type = *sentence->type();
+    std::transform(type.begin(), type.end(), type.begin(), [](unsigned char c) {
+        return std::toupper(c);
+    });
+    if (type == "ELASTICSEARCH") {
+        type_ = meta::cpp2::ListenerType::ELASTICSEARCH;
+    } else {
```
Comment (bright-starry-sky) [inline src/validator/AdminValidator.cpp:257]: I prefer to check on the graph level. WDYT?

**Review by bright-starry-sky — APPROVED** [review]
Well done.

File: src/validator/AdminValidator.cpp (hunk 3 above)
Comment (critical27) [reply src/validator/AdminValidator.cpp]: yeah, fixed

File: src/parser/parser.yy (hunk 2 above)
Comment (critical27) [reply src/parser/parser.yy]: fixed

File: src/executor/admin/ListenerExecutor.h (hunk 1 above)
Comment (critical27) [reply src/executor/admin/ListenerExecutor.h]: fixed

**Review by darionyaphet — APPROVED** [review]

**Pull request merged**